adcc/ExcitedStates.py

```diff
@@ -586,3 +654,60 @@ def describe_amplitudes(self, tolerance=0.01, index_format=None):
                 ret += "\n"
                 ret += separator
         return ret[:-1]
+
+    @property
+    def excitations(self):
+        """
+        Provides a list of Excitations, i.e., a view to all individual
+        excited states and their properties. Still under heavy development.
+        """
+        class Excitation:
+            pass
+
+        excitations = []
+        for i in range(self.size):
+            e = Excitation()
+            e.idx = i
+            e.__parent_state = self
+            for key in dir(self):
+                if key == "excitations":
+                    continue
+                if "__" in key or key.startswith("_"):
+                    continue  # skip "private" fields
+                if hasattr(type(self), key):
+                    if not isinstance(getattr(type(self), key), property):
+                        continue
+                else:
+                    continue
```

Comment:
I would branch differently. This I find very hard to read.

Reply:
True. Done.